source/kraken/tests/disruption_reader_test.cpp

```diff
@@ -690,3 +732,262 @@ BOOST_AUTO_TEST_CASE(two_properties) {
+    BOOST_REQUIRE_EQUAL(reader.impact->messages_size(), 1);
+    BOOST_CHECK_EQUAL(message->text(), "message_text_web");
+    BOOST_CHECK_EQUAL(message->created_at(), 1);
+    BOOST_CHECK_EQUAL(message->updated_at(), 2);
+
+    //Add a channel with a channel_type web
+    const_it.set_channel("channel_id", "channel_name_web", "type","400" ,"1", "2");
+    auto* channel = message->mutable_channel();
+    reader.fill_channel(const_it, channel);
+    BOOST_REQUIRE(message->mutable_channel() != nullptr);
+    BOOST_CHECK_EQUAL(channel->name(), "channel_name_web");
+    BOOST_CHECK_EQUAL(channel->content_type(), "type");
+    BOOST_CHECK_EQUAL(channel->max_size(), 400);
+    BOOST_CHECK_EQUAL(channel->created_at(), 1);
+    BOOST_CHECK_EQUAL(channel->updated_at(), 2);
+    const_it.set_channel_type("id_1", "web");
+    reader.fill_channel_type(const_it, channel);
+    BOOST_REQUIRE_EQUAL(channel->types_size(), 1);
+
+    // Add a message for mobile with channel and chanel_type
+    const_it.set_message("m_id_2", "message_text_mobile", "1", "2");
+    message = reader.impact->add_messages();
+    reader.fill_message(const_it, message);
+    BOOST_REQUIRE_EQUAL(reader.impact->messages_size(), 2);
+    BOOST_CHECK_EQUAL(message->text(), "message_text_mobile");
+    const_it.set_channel("channel_id_2", "channel_name_mobile", "type","400" ,"1", "2");
+    channel = message->mutable_channel();
+    reader.fill_channel(const_it, channel);
+    const_it.set_channel_type("id_2", "mobile");
+    reader.fill_channel_type(const_it, channel);
+    BOOST_REQUIRE_EQUAL(channel->types_size(), 1);
+    BOOST_REQUIRE_EQUAL(reader.impact->messages_size(), 2);
+
+    // Add a message for notification with channel and chanel_type
+    const_it.set_message("m_id_3", "message_text_notification", "1", "2");
+    message = reader.impact->add_messages();
+    reader.fill_message(const_it, message);
+    BOOST_REQUIRE_EQUAL(reader.impact->messages_size(), 3);
+    BOOST_CHECK_EQUAL(message->text(), "message_text_notification");
+    const_it.set_channel("channel_id_3", "channel_name_notification", "type","400" ,"1", "2");
+    channel = message->mutable_channel();
+    reader.fill_channel(const_it, channel);
+    const_it.set_channel_type("id_3", "notification");
+    reader.fill_channel_type(const_it, channel);
+    BOOST_REQUIRE_EQUAL(channel->types_size(), 1);
+}
+
+/* a badly sorted disruption record example
+ * Disruption record sorted in this way does not work for channel_type but works well for application_periods
+ * ..., message_1 , channel_1, channel_type_1, application_period_1 ...
+ * ..., message_2 , channel_2, channel_type_2, application_period_2 ...
+ * ..., message_3 , channel_3, channel_type_3, ...
+ * ..., message_1 , channel_1, channel_type_1, application_period_1 ...
+ * ..., message_2 , channel_2, channel_type_2, application_period_2 ...
+ * ..., message_3 , channel_3, channel_type_3, ...
+ */
+BOOST_AUTO_TEST_CASE(disruption_badly_sorted_informations) {
```

Comment:
This deserves more comment to explain what case you are testing, but are you testing here that it doesn't behave correctly? Do we want to enforce that behaviour?
To me it's interesting to see that you reproduiced the issue, but we cannot add it to tests.
Instead we should add a test that is using the correctly ordered db request, and that would crash if the order changes, because that's what you corrected.